spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalog.java

```diff
@@ -191,41 +279,50 @@ public boolean dropNamespace(String[] namespace, boolean cascade)
-    switch (provider.toLowerCase(Locale.ROOT)) {
-      case "hive":
-        return createAndInitSparkHiveCatalog(name, options);
-      default:
-        throw new NotSupportedException("Not support catalog: " + name + ", provider: " + provider);
+  private void validateNamespace(String[] namespace) {
+    Preconditions.checkArgument(
+        namespace.length == 1,
+        "Doesn't support multi level namespaces: " + String.join(".", namespace));
+  }
+
+  private String getDatabase(Identifier ident) {
+    if (ident.namespace().length > 0) {
+      return ident.namespace()[0];
     }
+    return getCatalogDefaultNamespace();
   }
 
-  private TableCatalog createAndInitSparkHiveCatalog(
-      String name, CaseInsensitiveStringMap options) {
+  private String getCatalogDefaultNamespace() {
+    String[] catalogDefaultNamespace = sparkCatalog.defaultNamespace();
     Preconditions.checkArgument(
-        gravitinoCatalogClient.properties() != null, "Hive Catalog properties should not be null");
-    String metastoreUri =
-        gravitinoCatalogClient.properties().get(GravitinoSparkConfig.GRAVITINO_HIVE_METASTORE_URI);
-    Preconditions.checkArgument(
-        StringUtils.isNotBlank(metastoreUri),
-        "Couldn't get "
-            + GravitinoSparkConfig.GRAVITINO_HIVE_METASTORE_URI
-            + " from hive catalog properties");
+        catalogDefaultNamespace != null && catalogDefaultNamespace.length == 1,
+        "Catalog default namespace is not valid");
+    return catalogDefaultNamespace[0];
+  }
 
-    TableCatalog hiveCatalog = new HiveTableCatalog();
-    HashMap<String, String> all = new HashMap<>(options);
-    all.put(GravitinoSparkConfig.SPARK_HIVE_METASTORE_URI, metastoreUri);
-    hiveCatalog.initialize(name, new CaseInsensitiveStringMap(all));
+  private com.datastrato.gravitino.rel.Column createGravitinoColumn(Column sparkColumn) {
+    return com.datastrato.gravitino.rel.Column.of(
+        sparkColumn.name(),
+        SparkTypeConverter.toGravitinoType(sparkColumn.dataType()),
+        sparkColumn.comment(),
+        sparkColumn.nullable(),
+        // Spark doesn't support autoIncrement
+        false,
+        // todo: support default value
+        com.datastrato.gravitino.rel.Column.DEFAULT_VALUE_NOT_SET);
+  }
 
-    return hiveCatalog;
+  private String[] getNamespace(NameIdentifier gravitinoIdentifier) {
+    Preconditions.checkArgument(
+        gravitinoIdentifier.namespace().length() == 3,
+        "Only support 3 level namespace," + gravitinoIdentifier.namespace());
+    return new String[] {gravitinoIdentifier.namespace().level(2)};
   }
 
-  private void validateNamespace(String[] namespace) {
+  private static void checkNotPathIdentifier(Identifier identifier, String method) {
     Preconditions.checkArgument(
-        namespace.length == 1,
-        "Doesn't support multi level namespaces: " + String.join(".", namespace));
+        !(identifier instanceof PathIdentifier),
```

Comment:
It is  a little weird that we depend the Iceberg here.

Reply:
will remove it.

Reply:
done